lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3395,6 +3395,153 @@ LogicalResult ConvertAtenOp<AtenGatherOp>::matchAndRewrite(
+  // 6]]) = torch.ops.aten.index_put(torch.tensor([[0, 1, 2, 3]]), # input
+  //                            (torch.tensor([0, 0, 0]), torch.tensor([1, 2,
+  //                            3])), # indicies torch.tensor([4, 5, 6])) #
+  //                            value
+  // = torch.ops.aten.index_put(torch.tensor([[0, 1, 2, 3]]), # input
+  //                            (None, torch.tensor([1, 2, 3]),),# indicies
+  //                            torch.tensor([4, 5, 6])) # value
+
+  // Not a tensor type.
+  auto input = adaptor.getSelf();
+  auto selfType = adaptor.getSelf().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
+
+  auto fillValues = adaptor.getValues();
+  auto valuesType = adaptor.getValues().getType().dyn_cast<TensorType>();
+  if (!valuesType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
+
+  // Deal with torch.prim.ListConstruct of non const value to get the index
+  auto tensorList = op.getIndices();
+  SmallVector<Value> tensorsTorchType;
+  if (!getListConstructElements(tensorList, tensorsTorchType))
+    return op.emitError(
+        "unimplemented: the tensor list is not from list construct");
+  auto indexTensors = getTypeConvertedValues(
+      rewriter, op->getLoc(), getTypeConverter(), tensorsTorchType);
+
+  auto outType = getTypeConverter()->convertType(op.getType());
+
+  // convert list of indices with none into indices tensor  without none
+  // indexTensors (none,[1,2,3]) -> ([0,0,0],[1,2,3])
+  // ([[0],[0],[0]],[[1],[2],[3]])-> [[0,1],[0,2], [0,3]]
+  if (indexTensors.size() <= 1) {
+    return rewriter.notifyMatchFailure(
+        op, "Only support indexput with multiple index.");
+  } else {
+    SmallVector<Value> indicesTfConcatTensors;
+    SmallVector<int64_t> indexesRank;
+    SmallVector<SmallVector<int64_t>> indexesShape;
+
+    // concat index tensor into to indices tensor for concat
+    for (size_t i = 0; i < indexTensors.size(); i++) {
+      auto index = indexTensors[i];
+      auto indexTorch = tensorsTorchType[i];
+      // TODO add support for none index other than i==0, like (index0, None)
+      // (None, index1)
+      if (i == 0 && indexTorch.getType().isa<Torch::NoneType>()) {
+        // convert None to [0,0,0]
+        auto indexNext = indexTensors[i + 1];
+        auto indexNextTorch = tensorsTorchType[i + 1];
+        if (indexNextTorch.getType().isa<Torch::NoneType>()){
+          return rewriter.notifyMatchFailure(
+              op, "Multiple None index is not support for now.");
+        }
+        auto indexNextType = indexNext.getType().dyn_cast<RankedTensorType>();
+        auto indexNextShape = indexNextType.getShape();
+
+        int64_t size = 1;
+        for (auto s : indexNextShape)
+          size *= s;
+        SmallVector<int32_t> values(size, i);
```

Comment:
Why is it correct to use a vector of zeros in the place where we had None?

Reply:
It is used to create indicesTf as `convertScatterNdOp` input. The algorithm inside `converScatterNdOp` is similar to the `convertGatherNdOp`   which will use these 0 to locate the flattened tensor.